ens/main.py

```diff
@@ -29,9 +45,30 @@
     raw_name_to_hash,
 )
 
+if TYPE_CHECKING:
+    from web3 import Web3  # noqa: F401
+    from web3.contract import (  # noqa: F401
+        Contract,
+    )
+    from web3.providers import (  # noqa: F401
+        BaseProvider,
+    )
+
+
 ENS_MAINNET_ADDR = '0x314159265dD8dbb310642f98f50C066173C1259b'
 
 
+# relocate / move to eth_typing
+class TxDict(TypedDict):
+    nonce: int
+    gasPrice: int
+    gas: int
+    # _from: Union[ChecksumAddress, str]  # address or ENS name
```

Comment:
I'm not sure what the best way to handle this is, since `TypedDict`s forbid using reserved keywords (`from`) as keys. Right now, I just `type: ignore` wherever is necessary - (i.e. line 171).

Reply:
I believe there is an alternate syntax for defining `TypedDict` that uses strings as field names.  Does it allow `from` there?

cc @kclowes maybe this is a breaking change we can work on introducing to change `from` to `sender`?

Reply:
Ahh found it, yup works fine now!

Reply:
Sounds good to me! Added to the v6 breaking changes issue!

Reply:
@kclowes Just my two cents - I got things working with `from` as a key, so I'm not sure it requires a breaking change to `sender`. Personally, I prefer `from` b/c of the relationship to the `to` key, but if we want to go ahead and update it to `sender` then maybe we should also update the `to` key to `receiver` or something to keep that relationship

Reply:
Sweet, I'm fine with either. I think both have equivalent pros and cons, but I like the idea of keeping the wording relationship with either `from/to` or `sender/receiver`. I'll remove the change from the breaking changes issue, but @pipermerriam if you have a strong opinion otherwise, let me know!